docs/modules/usage/configuration-options.md

```diff
@@ -0,0 +1,465 @@
+---
+id: configuration-options
+title: Configuration Options
+sidebar_label: Configuration Options
+---
+
+# Configuration Options
+
+This guide details all configuration options available for OpenHands, helping you customize its behavior and integrate it with other services. Required options and those with recommended settings are highlighted.
+
+---
+
+# Table of Contents
+
+1. [Core Configuration](#core-configuration)
+   - [API Keys](#api-keys)
+   - [Workspace](#workspace)
+   - [Debugging and Logging](#debugging-and-logging)
+   - [Session Management](#session-management)
+   - [Trajectories](#trajectories)
+   - [File Store](#file-store)
+   - [Task Management](#task-management)
+   - [Sandbox Configuration](#sandbox-configuration)
+   - [Miscellaneous](#miscellaneous)
+2. [LLM Configuration](#llm-configuration)
+   - [AWS Credentials](#aws-credentials)
+   - [API Configuration](#api-configuration)
+   - [Custom LLM Provider](#custom-llm-provider)
+   - [Embeddings](#embeddings)
+   - [Message Handling](#message-handling)
+   - [Model Selection](#model-selection)
+   - [Retrying](#retrying)
+   - [Advanced Options](#advanced-options)
+3. [Agent Configuration](#agent-configuration)
+   - [Microagent Configuration](#microagent-configuration)
+   - [Memory Configuration](#memory-configuration)
+   - [LLM Configuration](#llm-configuration-2)
+   - [ActionSpace Configuration](#actionspace-configuration)
+   - [Microagent Usage](#microagent-usage)
+4. [Sandbox Configuration](#sandbox-configuration-2)
+   - [Execution](#execution)
+   - [Container Image](#container-image)
+   - [Networking](#networking)
+   - [Linting and Plugins](#linting-and-plugins)
+   - [Dependencies and Environment](#dependencies-and-environment)
+   - [Evaluation](#evaluation)
+5. [Security Configuration](#security-configuration)
+   - [Confirmation Mode](#confirmation-mode)
+   - [Security Analyzer](#security-analyzer)
+
+
+---
+
+## Core Configuration
+The core configuration options are defined in the `[core]` section of the `config.toml` file.
+
+**API Keys**
+- `e2b_api_key`
+  - Type: `str`
+  - Default: `""`
+  - Description: API key for E2B
+
+- `modal_api_token_id`
+  - Type: `str` 
+  - Default: `""`
+  - Description: API token ID for Modal
+
+- `modal_api_token_secret`
+  - Type: `str`
+  - Default: `""`
+  - Description: API token secret for Modal
+
+**Workspace**
+- `workspace_base`
+  - Type: `str`
+  - Default: `"./workspace"`
+  - Description: Base path for the workspace
+
+- `cache_dir`
+  - Type: `str`
+  - Default: `"/tmp/cache"`
+  - Description: Cache directory path
+
+**Debugging and Logging**
+- `debug`
+  - Type: `bool`
+  - Default: `false`
+  - Description: Enable debugging
+
+- `disable_color`
+  - Type: `bool`
+  - Default: `false`
+  - Description: Disable color in terminal output
+
+**Session Management**
+- `enable_cli_session`
+  - Type: `bool`
+  - Default: `false`
+  - Description: Enable saving and restoring the session when run from CLI
```

Comment:
Just a small thing here, this does not exist anymore. It was removed sometime last month or so